src/main/java/no/nav/brukernotifikasjon/schemas/builders/OppgaveBuilder.java

```diff
@@ -0,0 +1,58 @@
+package no.nav.brukernotifikasjon.schemas.builders;
+
+import no.nav.brukernotifikasjon.schemas.Oppgave;
+import no.nav.brukernotifikasjon.schemas.builders.util.ValidationUtil;
+
+import java.net.URL;
+import java.time.LocalDateTime;
+
+public class OppgaveBuilder {
+
+    private LocalDateTime tidspunkt;
+    private String fodselsnummer;
+    private String grupperingsId;
+    private String tekst;
+    private URL link;
+    private Integer sikkerhetsnivaa;
+
+    public OppgaveBuilder withTidspunkt(LocalDateTime tidspunkt) {
+        this.tidspunkt = tidspunkt;
+        return this;
+    }
+
+    public OppgaveBuilder withFodselsnummer(String fodselsnummer) {
+        this.fodselsnummer = fodselsnummer;
+        return this;
+    }
+
+    public OppgaveBuilder withGrupperingsId(String grupperingsId) {
+        this.grupperingsId = grupperingsId;
+        return this;
+    }
+
+    public OppgaveBuilder withTekst(String tekst) {
+        this.tekst = tekst;
+        return this;
+    }
+
+    public OppgaveBuilder withLink(URL link) {
+        this.link = link;
+        return this;
+    }
+
+    public OppgaveBuilder withSikkerhetsnivaa(Integer sikkerhetsnivaa) {
+        this.sikkerhetsnivaa = sikkerhetsnivaa;
+        return this;
+    }
+
+    public Oppgave build() {
+        return new Oppgave(
+                ValidationUtil.localDateTimeToUtcTimestamp(tidspunkt, "tidspunkt", true),
+                ValidationUtil.validateFodselsnummer(fodselsnummer),
+                ValidationUtil.validateNonNullFieldMaxLength(grupperingsId, "grupperingsId", 100),
+                ValidationUtil.validateNonNullFieldMaxLength(tekst, "tekst", 500),
+                ValidationUtil.validateLink(link, "link", 200).toString(),
```

Comment:
Jeg tror vi må høre med Tone om Link feltet i Oppgave-eventet, for jeg tror ikke den skal kunne være null eller tom String. Husker Tone kommenterte dette fordi det er mulig å sende tom String i Aggregator, noe hun ikke trodde vi kunne 🤔

Reply:
Slik validateLink er implementert nå vil den kaste feil hvis link er null, eller lengre enn maxlength. Du tenker på at vi i tillegg bør legge til en sjekk på om den er tom?

Reply:
Ja, jeg tror vi må sjekke på tom string også, men jeg er ikke 100% sikker, så tror vi må snakke med Tone 😊

Reply:
Feltet sendes med uten verdi=tom, feltet sendes ikke med=null?
Jeg mener vi snakket om at det ikke gir noen mening at lenkefeltet er tomt/ikke har innhold. For at bruker skal kunne utføre en oppgave må den peke til noe. Vi har vel også sagt til teamene at det må være en lenke, så i praksis tror jeg ikke noen sender uten.

I doc https://navikt.github.io/brukernotifikasjon-docs/eventtyper/oppgave/felter/  står det "Produsenten kan sende inn en tom String. Da vil ikke link være synlig for sluttbruker." Er kanskje dette Vanessa refererer til med hvordan aggregator er i dag.

Reply:
Slik koden er nå, her i den nye builderen, vil eventet kunne opprettes med tom streng, men ikke med null. Altså ihht til det som står i dokumentasjonen.

Reply:
I denne nye builderen er link av datatype URL. Det vil ikke gå å sette den til tom streng, det aksepterer ikke den datatypen. Så skal man først sette den til noe annet enn null, må det være en syntaktisk korrekt URL. Slik jeg da forstår det skal regelen bli som følger:
Oppgave: Ikke tillatt at link er null.
Beskjed: Tillatt at link er null.
Statusoppdatering: Tillatt at link er null.

Enig @ToneEriksen @VanessaKjK ?

Reply:
Yass 💃 🎉